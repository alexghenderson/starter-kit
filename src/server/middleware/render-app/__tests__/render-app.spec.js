/**
 * @jest-environment node
 */
import fs from 'fs';
import renderApp from '../render-app';

jest.mock('fs', () => {
  return {
    readFileSync: jest.fn(() => '<script src="test.js"></script>'),
  };
});
jest.mock('~/components/root/server', () => () => 'test-markup');

describe('Render App Middleware', () => {
  it('should attempt to load index file by path', () => {
    renderApp('test-path')();
    expect(fs.readFileSync).toHaveBeenCalledWith('test-path', 'utf-8');
  });
  it('should replace index file html placeholder with rendered content', async () => {
    const handler = renderApp('test-path');
    const mockedSend = jest.fn();
    await handler({url: '/'}, {send: mockedSend});
    expect(mockedSend.mock.calls[0]).toMatchSnapshot();
  });
});
