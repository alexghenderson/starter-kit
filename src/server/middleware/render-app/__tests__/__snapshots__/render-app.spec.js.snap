// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Middleware should replace index file html placeholder with rendered content 1`] = `
Array [
  "<!DOCTYPE html><html data-reactroot=\\"\\"><head><title data-react-helmet=\\"true\\"></title><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"/><script type=\\"text/javascript\\">window.__INITIAL_STATE__={}</script></head><body><main id=\\"app\\">test-markup</main><script src=\\"test.js\\"></script></body></html>",
]
`;
